File: Easy/hidden_digits.py
data = ["abcdefghik", "Xa,}A#5N}{xOBwYBHIlH,#W", "(ABW>'yy^'M{X-K}q,", "6240488"]

Code = {'a':'0', 'b':'1', 'c':'2', 'd':'3', 'e':'4', 'f':'5', 'g':'6', 'h':'7', 'i':'8', 'j': '9',
		'0':'0', '1':'1', '2':'2', '3':'3', '4':'4', '5':'5', '6':'6', '7':'7', '8':'8', '9': '9'}

#import sys
#test_cases = open(sys.argv[1], 'r')
#data = test_cases.readlines()
for test in data:
	test = test.rstrip()
	HV_digits = ''
	for l in test:
		if l in Code:
			HV_digits = HV_digits + Code[l]
	if HV_digits == '':
		print 'NONE'
	else:
		print HV_digits

#test_cases.close()

File: Easy/hidden_digits.r
test.cases <- c("abcdefghik", "Xa,}A#5N}{xOBwYBHIlH,#W",
				"(ABW>'yy^'M{X-K}q,", "6240488")

#args <- commandArgs(trailingOnly=TRUE)
#test.cases <- strsplit(readLines(args[[1]], warn=FALSE), '\n')
nums <- as.character(c(0:9))
names(nums) <- letters[1:10]
cat(sapply(test.cases, function(test) {
	if (length(test) > 0) {
		chars <- strsplit(test, "")[[1]]
		chars <- chars[chars %in% names(nums) | chars %in% as.character(c(0:9))]
		result <- c()
		for (ch in chars) {
			if (ch %in% names(nums)) {
				ch <- nums[[ch]]
				result <- c(result, ch)
			} else {
				result <- c(result, ch)
			}
		}
		if (length(result) == 0) {
			"NONE"
		} else {
			paste(result, sep = "", collapse = "")
		}
	}
}), sep = "\n")